rotkehlchen/chain/ethereum/modules/eigenlayer/balances.py

```diff
@@ -0,0 +1,125 @@
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+
+from .constants import CPT_EIGENLAYER
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import EvmToken
+    from rotkehlchen.chain.ethereum.interfaces.balances import BalancesType
+    from rotkehlchen.chain.ethereum.node_inquirer import EthereumInquirer
+    from rotkehlchen.fval import FVal
+    from rotkehlchen.types import CHAIN_IDS_WITH_BALANCE_PROTOCOLS
+
+
+UNDERLYING_BALANCES_ABI: Final = [{'inputs': [], 'name': 'underlyingToken', 'outputs': [{'internalType': 'contract IERC20', 'name': '', 'type': 'address'}], 'stateMutability': 'view', 'type': 'function'}, {'inputs': [{'internalType': 'address', 'name': 'user', 'type': 'address'}], 'name': 'userUnderlyingView', 'outputs': [{'internalType': 'uint256', 'name': '', 'type': 'uint256'}], 'stateMutability': 'view', 'type': 'function'}]  # noqa: E501
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _read_underlying_assets(
+        evm_inquirer: 'EthereumInquirer',
+        strategy_address: ChecksumEvmAddress,
+        depositor: ChecksumEvmAddress,
+) -> tuple['FVal', 'EvmToken']:
+    """
+    Query the amount deposited in a eigenlayer strategy and the token of the strategy
+    May raise:
+    - RemoteError
+    - NotERC20Conformant
+    """
+    contract = EvmContract(
+        address=strategy_address,
+        abi=UNDERLYING_BALANCES_ABI,
+        deployed_block=0,
+    )
+    deposited_amount_raw = contract.call(
+        node_inquirer=evm_inquirer,
+        method_name='userUnderlyingView',
+        arguments=[depositor],
+    )
+    underlying_token_address = contract.call(
+        node_inquirer=evm_inquirer,
+        method_name='underlyingToken',
+    )
+    token = get_or_create_evm_token(
+        userdb=evm_inquirer.database,
+        evm_address=underlying_token_address,
+        chain_id=evm_inquirer.chain_id,
+    )
+    amount = asset_normalized_value(
+        amount=deposited_amount_raw,
+        asset=token,
+    )
+    return amount, token
+
+
+class EigenlayerBalances(ProtocolWithBalance):
+    def __init__(
+            self,
+            database: DBHandler,
+            evm_inquirer: 'EthereumInquirer',
+            chain_id: 'CHAIN_IDS_WITH_BALANCE_PROTOCOLS',
+    ):
+        super().__init__(
+            database=database,
+            evm_inquirer=evm_inquirer,
+            chain_id=chain_id,
+            counterparty=CPT_EIGENLAYER,
+            deposit_event_types={(HistoryEventType.STAKING, HistoryEventSubType.DEPOSIT_ASSET)},
+        )
+        self.evm_inquirer: 'EthereumInquirer'
+        self.web3 = self.evm_inquirer.web3_mapping
+        self.inquirer = Inquirer()
+
+    def query_balances(self) -> 'BalancesType':
+        """Query underlying balances for deposits in eigenlayer"""
```

Comment:
Write all potential exceptions this can raise.

For example I see you handle some remote errors but not remote errors raised by `inquirer.find_usd_price`